src/core/dynamic-user.c

```diff
@@ -607,21 +607,21 @@ static DynamicUser* dynamic_user_destroy(DynamicUser *d) {
         return dynamic_user_free(d);
 }
 
-int dynamic_user_serialize_one(DynamicUser *d, const char *key, FILE *f, FDSet *fds) {
+int dynamic_user_serialize_one(DynamicUser *d, const char *key, FILE *f, FDSet *fds, int **fds_array, size_t *n_fds_array) {
```

Comment:
hmm, I think we should really make FDSet an OrderedSet internally, and ten you can just fdset_size() as index to store. so you'd just specify in some flag or boolean or so which of the two serialization forms you want, but the data structure we pass in would always be the same.

Reply:
Let's hope we always get monotonically increasing FD numbers...